src/components/WowerlayRenderer.tsx

```diff
@@ -1,19 +1,109 @@
+import {
+  Direction,
+  PositionHandler,
+  PositionHandlerParameters,
+  checkOutOfScreenBottom,
+  checkOutOfScreenLeft,
+  checkOutOfScreenRight,
+  checkOutOfScreenTop,
+  getBottom,
+  getBottomEnd,
+  getBottomStart,
+  getLeft,
+  getLeftEnd,
+  getLeftStart,
+  getRight,
+  getRightEnd,
+  getRightStart,
+  getTop,
+  getTopEnd,
+  getTopStart
+} from './WowerlayUtils';
 import { cWowerlay, sWowerlayX, sWowerlayY } from '../consts';
 import { computed, defineComponent, onMounted, ref, watch } from 'vue';
 
+import { WowerlayProps } from './Wowerlay';
 import { useWowerlayContext } from '../event';
 import { wowerlayBaseProps } from './WowerlayReusables';
 
-enum Direction {
-  Horizontal,
-  Vertical
-}
+type Handlers = {
+  [Key in Exclude<WowerlayProps['position'], undefined>]: {
+    handle: PositionHandler;
+    handleOutOfScreen: PositionHandler;
+    checkOutOfScreen: (rect: PositionHandlerParameters) => boolean;
+  };
+};
+
+const positionHandlers: Handlers = {
+  top: {
+    handle: getTop,
+    handleOutOfScreen: getBottom,
+    checkOutOfScreen: checkOutOfScreenTop
```

Comment:
Can't the component itself do this check by position? As far as I can see, there is a repeating pattern.

Reply:
All positions should define when out of screen and where to go when out of screen. This pattern might be a little repetitive but gives so much flexibility.

Reply:
The code here does not reach the end user. It does not create any critical benefit within the developer. There is no need to retype the defaults over and over to add flexibility. It is sufficient to allow only to define custom calculations.

This can be refactored later. So you can mark it as resolved when other team members are approved.

Reply:
Actually in future we will add `custom-positions` option.

Reply:
Resolved in Meeting 🆗